### DISCUSSION TITLE
RA web User Enrollment Error on EJBCA 8.0

### DISCUSSION THREAD

**Discussion by merryo11** [Keyfactor/ejbca-ce#446] [General]
I created a role and access rule on PublicAccessAuthenticationToken with confidential transport (https) on  /ca_functionality/create_certificate and /ca_functionality/use_username. This rule is allowing us to access user enrollment link page over https without any admin/client certificate. 

Access link is: https://10.20.1.2/ejbca/ra/enrollusername.xhtml (Opening fine)

The issue is when I try to enroll any entity through it to get the certificate I am not able to do and getting below error.

![userenroll-error](https://github.com/Keyfactor/ejbca-ce/assets/144104956/313bc7c7-22dd-4de3-a639-a52f0f6f88f0)

After some checking I found out that when we access the same page as a authenticated user my enrollment gets successful.

So any solution to this shall be very helpful

**Comment by primetomas** [comment]
You have a guide for public access here.
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ra-operations-guide/configure-ejbca-for-public-access

**Comment by merryo11** [comment]
Thank you thomas the link was indeed very helpful and resolved my issue.
